frontend/src/components/SignIn.tsx

```diff
@@ -35,8 +38,32 @@ export default function SignIn() {
     },
   });
 
+  const navigate = useNavigate();
+
+  const { mutate } = useMutation({
```

Comment:
An idea would be to separate these hooks into custom ones to simplify the usage on the components.

`useSignIN()`, `useSignUp` etc